docs/edgeql/path_resolution.rst

```diff
@@ -0,0 +1,151 @@
+.. _ref_eql_path_resolution:
+
+===============
+Path resolution
+===============
+
+Element-wise operations with multiple arguments in EdgeDB are generally applied
+to the :ref:`cartesian product <ref_reference_cardinality_cartesian>` of all
+the input sets.
+
+.. code-block:: edgeql-repl
+
+    db> select {'aaa', 'bbb'} ++ {'ccc', 'ddd'};
+    {'aaaccc', 'aaaddd', 'bbbccc', 'bbbddd'}
+
+In some cases, this works out fine, but in others it doesn't make sense. Take
+this example:
+
+.. code-block:: edgeql
+
+    select User.first_name ++ ' ' ++ User.last_name;
+
+Should the result of this query be every ``first_name`` in your ``User``
+objects concatenated with every ``last_name``? That's probably not what you
+want.
+
+This is why, in cases where multiple element-wise arguments share a common path
+(``User.`` in this example), EdgeDB factors out the common path rather than
+using cartesian multiplication.
+
+.. code-block:: edgeql-repl
+
+    db> select User.first_name ++ ' ' ++ User.last_name;
+    {'Mina Murray', 'Jonathan Harker', 'Lucy Westenra', 'John Seward'}
+
+We assume this is what you want, but if your goal is to get the cartesian
+product, you can accomplish it one of two ways. You could use
+:eql:op:`detached`.
```

Comment:
There's a third way, too, which is to put both parts in a subquery like in the "Scopes" section